From f5009febf4da9b130702dbcd33b5e4869014b490 Mon Sep 17 00:00:00 2001
From: Amelie Delaunay <amelie.delaunay@foss.st.com>
Date: Tue, 29 Jul 2025 11:27:38 +0200
Subject: [PATCH] arm64: dts: st: add I3C4 support on stm32mp257f-ev1

I3C4 is available on GPIO expansion connector and on mikroBus connector of
stm32mp257f-ev1 board.
On GPIO expansion connector and on mikroBus connector, it can use the same
pins as I2C8 (PZ3 and PZ4 configured with i3c4 pins_a pinctrl), but there
are external pull-ups on the board (it affects the performance of the I3C).
On mikroBus connector, it could also be configured with i3c4 pins_b pinctrl
but it uses the same pins as SPI8, without external pull-up on the board.

Change-Id: I6f9ec764e2a575ffb33492813c46d8dde5b22b68
Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/478606
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Alain VOLMAT <alain.volmat@st.com>
---
 arch/arm64/boot/dts/st/stm32mp257f-ev1.dts | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts b/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
index 81da498a6401..4b0445a77610 100644
--- a/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
@@ -466,6 +466,23 @@ i2s2_endpoint: endpoint {
 	};
 };
 
+&i3c4 {
+	/*
+	 *  Connector | pins_a  | pins_b |
+	 * -----------+---------+--------|
+	 *  GPIO Exp. | I2C8+PU |   X    |
+	 * -----------+---------+--------|
+	 *  mikroBus  | I2C8+PU | SPI8   |
+	 *
+	 * X = not wired ; PU = External Pull-Up
+	 */
+	pinctrl-names = "default", "init", "sleep";
+	pinctrl-0 = <&i3c4_pins_b>;
+	pinctrl-1 = <&i3c4_init_pins_b>;
+	pinctrl-2 = <&i3c4_sleep_pins_b>;
+	status = "disabled";
+};
+
 &ipcc1 {
 	status = "okay";
 };
-- 
2.53.0

